comp/core/tagger/impl/tagger.go

```diff
@@ -0,0 +1,552 @@
+// standard tags (env, version, service) from cache or sources.
+func (t *TaggerWrapper) Standard(entityID types.EntityID) ([]string, error) {
+	t.mux.RLock()
+	// TODO(components) (tagger): captureTagger is a legacy global variable to be eliminated
+	if t.captureTagger != nil {
+		tags, err := t.captureTagger.Standard(entityID)
+		if err == nil && len(tags) > 0 {
+			t.mux.RUnlock()
+			return tags, nil
+		}
+	}
+	t.mux.RUnlock()
+	return t.defaultTagger.Standard(entityID)
+}
+
+// AgentTags returns the agent tags
+// It relies on the container provider utils to get the Agent container ID
+func (t *TaggerWrapper) AgentTags(cardinality types.TagCardinality) ([]string, error) {
+	ctrID, err := metrics.GetProvider(t.wmeta).GetMetaCollector().GetSelfContainerID()
+	if err != nil {
+		return nil, err
+	}
+
+	if ctrID == "" {
+		return nil, nil
+	}
+
+	entityID := types.NewEntityID(types.ContainerID, ctrID)
+	return t.Tag(entityID, cardinality)
+}
+
+// GlobalTags queries global tags that should apply to all data coming from the
+// agent.
+func (t *TaggerWrapper) GlobalTags(cardinality types.TagCardinality) ([]string, error) {
+	t.mux.RLock()
+	if t.captureTagger != nil {
+		tags, err := t.captureTagger.Tag(taggercommon.GetGlobalEntityID(), cardinality)
+		if err == nil && len(tags) > 0 {
+			t.mux.RUnlock()
+			return tags, nil
+		}
+	}
+	t.mux.RUnlock()
+	return t.defaultTagger.Tag(taggercommon.GetGlobalEntityID(), cardinality)
+}
+
+// globalTagBuilder queries global tags that should apply to all data coming
+// from the agent and appends them to the TagsAccumulator
+func (t *TaggerWrapper) globalTagBuilder(cardinality types.TagCardinality, tb tagset.TagsAccumulator) error {
+	t.mux.RLock()
+	if t.captureTagger != nil {
+		err := t.captureTagger.AccumulateTagsFor(taggercommon.GetGlobalEntityID(), cardinality, tb)
+
+		if err == nil {
+			t.mux.RUnlock()
+			return nil
+		}
+	}
+	t.mux.RUnlock()
+	return t.defaultTagger.AccumulateTagsFor(taggercommon.GetGlobalEntityID(), cardinality, tb)
+}
+
+// List the content of the defaulTagger
+func (t *TaggerWrapper) List() types.TaggerListResponse {
+	return t.defaultTagger.List()
+}
+
+// SetNewCaptureTagger sets the tagger to be used when replaying a capture
+func (t *TaggerWrapper) SetNewCaptureTagger(newCaptureTagger tagger.Component) {
+	t.mux.Lock()
+	defer t.mux.Unlock()
+	t.captureTagger = newCaptureTagger
+}
+
+// ResetCaptureTagger resets the capture tagger to nil
+func (t *TaggerWrapper) ResetCaptureTagger() {
+	t.mux.Lock()
+	defer t.mux.Unlock()
+	t.captureTagger = nil
+}
+
+// EnrichTags extends a tag list with origin detection tags
+// NOTE(remy): it is not needed to sort/dedup the tags anymore since after the
+// enrichment, the metric and its tags is sent to the context key generator, which
+// is taking care of deduping the tags while generating the context key.
+func (t *TaggerWrapper) EnrichTags(tb tagset.TagsAccumulator, originInfo taggertypes.OriginInfo) {
```

Comment:
Should this logic be in `local_tagger.go` instead?
I think the idea of this wrapper is that it includes a local tagger and the replay. So I'd expect all methods to be something like:
```
if t.captureTagger != nil {
  // call capture tagger
}
// call default tagger
```

So I think all the logic should be implemented in local_tagger.go which is already the case in other functions of this file.

Reply:
If you look at how is currently in the codebase, `EnrichTags` does not check for `captureTagger` https://github.com/DataDog/datadog-agent/blob/main/comp/core/tagger/taggerimpl/tagger.go#L403-L548

So I prefer not to add new logic to this PR to ensure the behavior remains the same. We could see if we need to use the `captureTagger` for `EnrichTags` on a separate PR

Reply:
Sorry, my message was not clear. I didn't meant to say that we should check the captureTagger in this function. What I wanted to say is that the content of the function could be moved to local_tagger.go, and this one should be `t.defaultTagger.EnrichTags(...)`

Reply:
> and this one should be t.defaultTagger.EnrichTags(...)